rust-runtime/aws-smithy-runtime-api/src/http/headers.rs

```diff
@@ -92,28 +92,31 @@ impl Headers {
     /// This will *replace* any existing value for this key. Returns the previous associated value if any.
     ///
     /// # Panics
-    /// If the key or value are not valid ascii, this function will panic.
+    /// If the key or value are not valid ASCII, this function will panic.
```

Comment:
the panic conditions are actually more subtle (although maybe we should eschew using `from_static` at all (or only use it if possible, but all the header to be made in the dynamic way?)

we could have the logic be:
- if we have a static value and it's all lower case and ready to go: use `from_static`
- otherwise try to go through the `HeaderName` constructor

seems like this would do the right thing for most use cases. We could emit a trace log if someone gave us a static header and we couldn't make use of it because, e.g. it was upper case.